skimage/color/colorconv.py

```diff
@@ -804,6 +804,65 @@ def rgb2grey(rgb):
     return rgb2gray(rgb)
 
 
+def gray2rgba(image, alpha=None):
+    """Create a RGBA representation of a gray-level image.
+
+    Parameters
+    ----------
+    image : array_like
+        Input image.
+    alpha : array_like, optional
+        Alpha channel of the output image. It may be a scalar or an
+        array with the same shape as ``image``. If not specified it is
+        set to the maximum limit correspondin to ``image`` data type.
+
+    Returns
+    -------
+    rgba : ndarray
+        RGBA image. A new dimension of length 4 is added to input
+        image shape.
+
+    Raises
+    ------
+    ValueError
+        If ``alpha`` is a scala and is not in ``image`` data type
+        limits.
+    ValueError
+        If ``alpha`` is an array and have a different shape then
+        ``image``.
+    ValueError
+        If ``alpha`` is an array and have a different data type then
+        ``image``.
+
+    """
+
+    arr = np.asarray(image)
+
+    alpha_min, alpha_max = dtype_limits(arr, clip_negative=False)
+
+    if alpha is None:
+        alpha = alpha_max
+    elif np.isscalar(alpha):
+        if alpha_min > alpha or alpha > alpha_max:
```

Comment:
I don't think this check is necessary for floating point numbers.

For scalars, maybe.

Reply:
For floating point numbers, `dtype_limits` are [-1, 1]...
Even if we don't want to restrict images range, we have to be careful about the meaning of an alpha channel. Even if the "junk in -> junk out" credo is reasonable, the user must at least be warned that he provided what we consider junk ;-)

Reply:
I guess, now that we've decided to loosen up on floating point limits in general, this is a problematic situation.  I don't think there is a "correct" solution now, but limiting the alpha value specified may go against what the user wants.

Reply:
Do you mean that this check is no more necessary?